### PR TITLE
Header20

### PR DESCRIPTION
**Pull request by yapmakutu**

### CODE REVIEW

**Review by Ayazelar — COMMENTED** [review]
So many "if case" used, it should be reduced

File: editor-components/header/header20/header20.tsx [hunk 1]
```diff
@@ -2,341 +2,1033 @@ import * as React from "react";
...
+              value: "09",
+            },
             {
               type: "image",
-              key: "image",
-              value: "https://cdn-icons-png.flaticon.com/512/145/145808.png",
               displayer: "Image",
+              key: "image",
+              value:
+                "http://clapat.ro/themes/hervin-wordpress/wp-content/uploads/2019/05/09hero.jpg",
```
Comment (dogusmiuw) [inline editor-components/header/header20/header20.tsx:528]: use spica storage url

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

File: editor-components/header/header20/header20.module.scss [hunk 2]
```diff
@@ -1,223 +1,359 @@
...
-            font-size: 30px;
           }
-          .button-and-input {
-            .input {
-              padding: 5px 20px;
+          .icon-text-container {
+            justify-content: center;
+            display: flex;
+            width: auto;
+            // max-width: fit-content;
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.module.scss:188]: remove unused style

File: editor-components/header/header20/header20.module.scss [hunk 3]
```diff
@@ -1,223 +1,359 @@
...
-                  height: 20px;
+
+            &:hover .icon-next-to-text {
+              transform: translate(0, -4rem);
+              opacity: 0;
+              transition: all 0.3s cubic-bezier(0.215, 0.61, 0.355, 1) 0s;
+            }
+
+            .icon_text {
+              transition: all 0.3s cubic-bezier(0.215, 0.61, 0.355, 1) 0.3s;
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.module.scss:210]: use varaible for transition

File: editor-components/header/header20/header20.tsx [hunk 4]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+
+
+
+  debounce = <T extends (...args: any[]) => void>(func: T, delay: number): ((...args: Parameters<T>) => void) => {
+    let timeoutId: NodeJS.Timeout;
+    return (...args: Parameters<T>) => {
+      clearTimeout(timeoutId);
+      timeoutId = setTimeout(() => func(...args), delay);
+    };
+  };
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:873]:
```suggestion
```

File: editor-components/header/header20/header20.tsx [hunk 5]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+        func(...args);
+        inThrottle = true;
+        setTimeout(() => (inThrottle = false), limit);
+      }
+    };
+  };
+
+
+
+  debounce = <T extends (...args: any[]) => void>(func: T, delay: number): ((...args: Parameters<T>) => void) => {
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:867]: delete debounce function cause unused

File: editor-components/header/header20/header20.tsx [hunk 6]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+    let timeoutId: NodeJS.Timeout;
+    return (...args: Parameters<T>) => {
+      clearTimeout(timeoutId);
+      timeoutId = setTimeout(() => func(...args), delay);
+    };
+  };
+
+  handleWheel = this.throttle((event: React.WheelEvent) => {
+    if (event.deltaY < 0) {
+      console.log("yukarı");
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:877]: remove log

File: editor-components/header/header20/header20.tsx [hunk 7]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+
+
+
 
   getName(): string {
     return "Header-20";
   }
 
+  handleUpClick = () => {
+    console.log("up")
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:893]: remove log

File: editor-components/header/header20/header20.tsx [hunk 8]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+    const currentSlide = this.getComponentState("slider");
+    const nextSlide = Math.max(currentSlide - 1, 0);
+    console.log("next slider", nextSlide)
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:896]: remove log

File: editor-components/header/header20/header20.tsx [hunk 9]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+    if (this.titleSliderRef.current) {
+      this.titleSliderRef.current.slickGoTo(nextSlide);
+    }
+    if (this.commentSliderRef.current) {
+      this.commentSliderRef.current.slickGoTo(nextSlide);
+    }
+
+    this.setComponentState("slider", nextSlide);
+    this.setComponentState("titleSlider", nextSlide);
+    this.setComponentState("commentSlider", nextSlide);
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:910]: create a function for this cause you are using line 921

File: editor-components/header/header20/header20.tsx [hunk 10]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+                            e.currentTarget.style.transform = "scale(1.2)";
+                            e.currentTarget.style.fontWeight = "bold";
+                          }}
+                          onMouseLeave={(e) => {
+                            e.currentTarget.style.transform = "scale(1)";
+                            e.currentTarget.style.fontWeight = "normal";
+                          }}
+                        >
+                          {icon.icon_text}
+                        </a>
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:1142]: Use ComposerLink and remove onMouseEnter and onMouseLeave.

File: editor-components/header/header20/header20.tsx [hunk 11]
```diff
@@ -2,343 +2,1156 @@ import * as React from "react";
...
+      titlesToShow.push({ ...slider[index], position: "current" });
+      if (slider[index + 1])
+        titlesToShow.push({ ...slider[index + 1], position: "next" });
+    }
+
+    return titlesToShow.map((title) => ({
+      ...title,
+      link: "link" in title ? title.link : undefined,
+    }));
+  };
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:969]: check this function seem unnessasary

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

**Review by Ayazelar — APPROVED** [review]

File: editor-components/header/header20/header20.tsx [hunk 12]
```diff
@@ -2,343 +2,1091 @@ import * as React from "react";
...
+                        name={slider[currentSlide].buttomRow.icon}
+                        propsIcon={{
+                          className: this.decorateCSS("icon-next-to-text"),
+                        }}
+                      />
+                    )}
+                    <div className={this.decorateCSS("social-icons")}>
+                      {icons.map((icon, i) => (
+                        <div className={this.decorateCSS("icon")}>
+                          <ComposerLink>
```
Comment (Ayazelar) [inline editor-components/header/header20/header20.tsx:1074]: give path

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

**Review by Ayazelar — APPROVED** [review]

**Pull request merged**